File: 1550_ThreeConsecutiveOdds.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool threeConsecutiveOdds(vector<int>& arr) {
        int counter=0;
        for (auto& n: arr){
            if (n%2==0){
                counter=0;
            } else {
                if (counter==2){
                    return true;
                }
                ++counter;
            }
        }
        return false;
    }
};

File: 1550_ThreeConsecutiveOdds.py
from typing import List

class Solution:
    def threeConsecutiveOdds(self, arr: List[int]) -> bool:
        oddCount=0
        for num in arr:
            if num%2:
                oddCount+=1
                if oddCount==3:
                    return True
            else:
                oddCount=0
        
        return False